src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -13,25 +13,33 @@
 # limitations under the License.
 
 import asyncio
+import gc
 import logging
-import unittest
 import time
-import gc
+import unittest
 
 import grpc
 from grpc.experimental import aio
-from tests_aio.unit._test_base import AioTestBase
+
 from tests.unit.framework.common import test_constants
+from tests_aio.unit._test_base import AioTestBase
 
 _SIMPLE_UNARY_UNARY = '/test/SimpleUnaryUnary'
 _BLOCK_FOREVER = '/test/BlockForever'
 _BLOCK_BRIEFLY = '/test/BlockBriefly'
 _UNARY_STREAM_ASYNC_GEN = '/test/UnaryStreamAsyncGen'
 _UNARY_STREAM_READER_WRITER = '/test/UnaryStreamReaderWriter'
 _UNARY_STREAM_EVILLY_MIXED = '/test/UnaryStreamEvillyMixed'
+_STREAM_UNARY_ASYNC_GEN = '/test/StreamUnaryAsyncGen'
+_STREAM_UNARY_READER_WRITER = '/test/StreamUnaryReaderWriter'
+_STREAM_UNARY_EVILLY_MIXED = '/test/StreamUnaryEvillyMixed'
```

Comment:
Mixing two styles might have unexpected behavior due to race between them.

For iterators, users are trusting our library to consume them. But for reader / writer API, the manner of message consumption is controlled by users.

E.g. if a user supplied a request iterator, and then invoking `write` at another coroutine. If the request iterator depleted, it will trigger `done_writing`, then the other coroutine will fail seems randomly.